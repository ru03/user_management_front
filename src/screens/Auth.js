import React, { useEffect } from 'react';
import { Formik } from 'formik';
import { Redirect } from 'react-router-dom';
import { Container } from '@material-ui/core';
import { loginValidation } from '../utils/validations'
import useFetch from '../hooks/fetch';
import env from '../config/config';
import { Alert } from '../components/UI';
import Login from '../components/forms/auth/Login';

const Auth = () => {
  const { data, error, sendRequest } = useFetch();

  useEffect(() => {
    if (data) localStorage.setItem('token', data.token);
  }, [data]);

  const onSubmit = (values, helpers) => {
    const config = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(values),
    }
    sendRequest(`${env.basepath}${env.auth}`, config);
    helpers.setSubmitting(false);
  }
  return (
    <Container>
      <Alert message={error} />
      {
        data
          ? <Redirect to="/" />
          : <Formik
            initialValues={{
              email: '',
              password: '',
            }}
            validate={loginValidation}
            onSubmit={onSubmit}
          >
            {props => <Login {...props} />}
          </Formik>
      }
    </Container>
  )
};

export default Auth;