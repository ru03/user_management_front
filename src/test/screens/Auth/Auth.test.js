import React from 'react';
import { render } from '@testing-library/react';
import AuthProvider from '../../../providers/authProvider';
import Auth from '../../../screens/Auth';

describe('Auth', () => {
  it('renders', () => {
    const { asFragment } = render(<AuthProvider><Auth /></AuthProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});