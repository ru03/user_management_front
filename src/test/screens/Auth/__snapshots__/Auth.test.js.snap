// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth renders 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <form
      data-testid="loginForm"
    >
      <div
        class="MuiBox-root MuiBox-root-12"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-10 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardHeader-root makeStyles-header-9"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  Login
                </span>
              </div>
            </div>
            <div
              class="MuiCardContent-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                    for="email"
                    id="email-label"
                  >
                    Email
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      id="email"
                      name="email"
                      placeholder="email"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                    for="password"
                    id="password-label"
                  >
                    Password
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      id="password"
                      name="password"
                      placeholder="password"
                      type="password"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                  data-testid="buttonSubmit"
                  id="login_button"
                  tabindex="0"
                  type="submit"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Login
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
